hoomd/md/compute.py

```diff
@@ -0,0 +1,276 @@
+# Copyright (c) 2009-2019 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+# Maintainer: joaander / All Developers are free to add commands for new features
+
+r""" Compute system properties
+
+A compute calculates properties of the system on demand. Most computes are automatically created by the object
+that needs them (e.g. methods.NVT creates a compute.ThermodynamicQuantities for temperature calculations). User-specified
+computes can be used when more flexibility is needed. Properties calculated by specified computes (automatically, or by the
+user) can be logged using hoomd.logging.Logger.
+"""
+
+from hoomd import _hoomd;
+from hoomd.md import _md
+from hoomd.operation import _Compute
+from hoomd.logging import log
+import hoomd;
+import sys;
+
+class _Thermo(_Compute):
+
+    def __init__(self, filter):
+        self._filter = filter
+
+
+class ThermodynamicQuantities(_Thermo):
+    R""" Compute thermodynamic properties of a group of particles.
+
+    Args:
+        filter (``hoomd.filter``): Particle filter to compute thermodynamic properties for.
+
+    :py:class:`ThermodynamicQuantities` acts on a given group of particles and calculates thermodynamic properties of
+    those particles when requested. All specified :py:class:`ThermodynamicQuantities` objects can be added to a logger
+    for logging during a simulation, see :py:class:`hoomd.logging.Logger` for more details.
+
```

Comment:
Let's add a note here about pairwise interactions:
```
Note:

    For pairwise terms U_ij and F_ij \cdot r_ij), HOOMD-blue assigns half of each term to a per particle quantity. When *filter* selects a subset of the particles in the system, `ThermoDynamicQuantities` sums these per-particle assignments over only particles selected by the filter.
```
What do people think? Is that clear enough, or should we instead provide a full LaTeX math equation for U and P that include this explicitly?

Reply:
I think the math equation would make it clearer what is happening, especially why it assigns half of each term to a per particle quantity. Do users on the email list ask about this often?

Reply:
No, it is not a common question, but it is good to be explicit.

Reply:
OK. In the pressure equations we can note that the i and j indicies are particle tags and write the sums over `k \in filter`.

For the potential energy U, we would write a similar expression but break it into components. `U=U_pair + U_bond + U_angle + ...` and then explicitly document the expressions for the terms. Something like `U_pair = sum over i \in filter 1/2 V_ij`

Reply:
I don't quite understand what you mean for the pressure (`P`) equations. Why would we write the sums over `k` (and presumably `k'`) if we want to document the usage of `i` and `j`?

Reply:
You are talking about the sums in the equation for `W`, right?

Reply:
I was speaking in general terms. All equations that sum over pairs will have some indices that run over the filtered subset and others that run over all particles.